opal/mca/btl/tcp/btl_tcp_component.c

```diff
@@ -1183,8 +1183,8 @@ static int mca_btl_tcp_component_exchange(void)
                  if ((AF_INET == my_ss.ss_family) &&
                      (4 != mca_btl_tcp_component.tcp_disable_family)) {
                      memcpy(&addrs[current_addr].addr_inet,
-                             &((struct sockaddr_in*)&my_ss)->sin_addr,
-                             sizeof(addrs[0].addr_inet));
+                            &((struct sockaddr_in*)&my_ss)->sin_addr,
+                            sizeof(((struct sockaddr_in*)&my_ss)->sin_addr));
```

Comment:
How is that not the same ?

Reply:
sizeof(addrs[0].addr_inet) == 16.
sizeof(((struct sockaddr_in*)&my_ss)->sin_addr) == 4

Reply:
Right. How about sizeof(struct in_addr) to make out intentions clear.

Reply:
I admit that I kinda like the style of `memcpy(dest, &src, sizeof(src))`.  Do you think that's not clear?

Reply:
In other places we use "sizeof(struct in_addr)", which I think is a good indicator that we are dealing with an IPv4 address.

As a special mention, the original code was correct, we are moving 16 bytes around, but only the sin_addr part is initialized.

Reply:
Updated to `sizeof(struct in_addr)`.